core/src/main/java/org/elasticsearch/index/query/FilteredQueryParser.java

```diff
@@ -83,44 +78,15 @@ public Query parse(QueryParseContext parseContext) throws IOException, QueryPars
             }
         }
 
-        // parsed internally, but returned null during parsing...
-        if (query == null) {
-            return null;
-        }
-
-        if (filter == null) {
-            if (!filterFound) {
-                // we allow for null filter, so it makes compositions on the client side to be simpler
-                return query;
-            } else {
-                // even if the filter is not found, and its null, we should simply ignore it, and go
-                // by the query
-                return query;
-            }
-        }
-        if (Queries.isConstantMatchAllQuery(filter)) {
-            // this is an instance of match all filter, just execute the query
-            return query;
-        }
-
-        // if its a match_all query, use constant_score
-        if (Queries.isConstantMatchAllQuery(query)) {
-            Query q = new ConstantScoreQuery(filter);
-            q.setBoost(boost);
-            return q;
-        }
-
-        BooleanQuery filteredQuery = Queries.filtered(query, filter);
-
-        filteredQuery.setBoost(boost);
-        if (queryName != null) {
-            parseContext.addNamedQuery(queryName, filteredQuery);
-        }
-        return filteredQuery;
```

Comment:
what doesn't convince me about removing all these checks is that there is a difference between returning null and returning match_all I think, which we had before. null means ignore (no-op), while match_all is a query that matches all documents. Also this is going to change again with #11915. I would rather wait, get #11915 in and then refactor this one once and forever :)